src/Parser.php

```diff
@@ -60,14 +60,22 @@ public function validate(bool $useSchema = true): bool
 
     protected function libXmlErrorToException($error): XMLException
     {
+        if (
```

Comment:
Perhaps this could live as a static method on the `XmlException` class (e.g. `parseInvalidChar()`). And we could parse it from the `$message` parameter on the exception. That way we don't need to tack on an extra parameter to the exception constructor too.